pumpkin/src/entity/to_packet.rs

```diff
@@ -0,0 +1,28 @@
+use crate::entity::Entity;
+use pumpkin_protocol::client::play::CSpawnEntity;
+
+pub fn spawn_entity_with_data(entity: &Entity, data: Option<i32>) -> CSpawnEntity {
+    let pos = entity.pos.load();
+    let velocity = entity.velocity.load();
+    CSpawnEntity::new(
+        entity.entity_id.into(),
+        entity.uuid,
+        (entity.entity_type as i32).into(),
+        pos.x,
+        pos.y,
+        pos.z,
+        entity.pitch.load(),
+        entity.yaw.load(),
+        entity.head_yaw.load(),
+        data.unwrap_or(0).into(),
+        velocity.x,
+        velocity.y,
+        velocity.z,
+    )
+}
+
+impl From<&Entity> for CSpawnEntity {
+    fn from(entity: &Entity) -> Self {
+        spawn_entity_with_data(entity, None)
+    }
+}
```

Comment:
I really don't like this, First you have a whole new file for only wrapping around some values, Also i don't think its that hard to write this every time, If you think this is really necessary i would much more prefer this in the entity file